### ISSUE TITLE
[Bug] Default user agent style sheet colors all `a` elements.

### ISSUE THREAD

**Issue by brunnerh** [darkreader/darkreader#9836]
# Bug Report

## Website Address
Not related to a specific page.

## Bug Description
An `a` element is only a link if it has an `href` Attribute. Browsers will not color them if they are not links.
Chrome uses `a:-webkit-any-link` to select them by default.

`<style class="darkreader darkreader--user-agent">` has a plain `a` selector instead of `a:any-link` which causes non-links to turn blue.

## Steps To Reproduce
Create a new file or go to a site like Codepen and add
```html
<a href="/">A valid link</a><br/>
<a>Not a valid link</a>
```

## Expected Behavior
The element that is not a link is not colored.

## Actual Behavior
The element that is not a link is colored.

## Screenshots
Without darkreader:
![image](https://user-images.githubusercontent.com/834235/190667509-c4321bd3-60b4-4b76-abce-e484c18c5c98.png)

With darkreader:
![image](https://user-images.githubusercontent.com/834235/190667598-db984c5c-01dc-43b1-aa77-8f36f4daae88.png)

## System Information

- OS: Windows 10
- Browser: Chrome 105
- Dark Reader Version: 4.9.57

## Additional Context
None

**Comment by brunnerh** [comment]
Workaround CSS rule:
```css
[data-darkreader-scheme="dark"] a:not(:any-link) {
	color: unset;
}
```

**Comment by Gusted** [comment]
Hi @brunnerh!

Thanks for your bug report, this is indeed a bug and should be solved now. Quite funny that this seems to never really broken any real-world cases.

**Comment by brunnerh** [comment]
Well, there usually is no reason to have an `a` that is not a link.

I just ran into this because i was lazy and "deactivated" a link by simply removing the `href` rather than swapping out the element type.

**Comment by brunnerh** [comment]
Isn't this just a specificity issue that could be solved with:

```css
a:where(:any-link) { ... }
```

**Comment by Gusted** [comment]
That's a option, for now just holding off to change that bit of code again and gather some real world tests.

**Comment by bershanskiy** [comment]
> That's a option, for now just holding off to change that bit of code again and gather some real world tests.

Agreed, let's hold out on this change until after the next release (which should happen in the next few days).

**Comment by Gusted** [comment]
Relevant info in #10442
